File: Expense_BE/src/main/java/com/hunglevi/expense_be/model/Budget.java
package com.hunglevi.expense_be.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "budget")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Budget{
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private Double amount;
        private String period;
        private String createdAt;
        @OneToOne
        @JoinColumn(name = "user_id")
        private User user;
        @ManyToOne
        @JoinColumn(name = "category_id")
        private Category category;

}

File: Expense_MDC/app/src/main/java/com/hunglevi/expense_mdc/data/model/Budget.kt
package com.hunglevi.expense_mdc.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "budgets")
data class Budget(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,           // Auto-increment primary key
    val userId: Int,          // User ID
    val categoryId: Int,      // Transaction category
    val amount: Double,        // Transaction amount
    val period : String,         // Budget period (e.g., monthly, yearly)
    val createdAt: String
)